Frontend/Gmfctn/src/app/core/interceptors/http/main-http.interseptor.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpErrorResponse, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { Observable, of, throwError } from 'rxjs';
+import { catchError, switchMap } from 'rxjs/operators';
+import { AuthenticateService } from 'src/app/modules/+auth/services/authenticate.service';
+
+
+@Injectable()
+export class MainHttpInterceptor implements HttpInterceptor {
```

Comment:
MainHttpInterceptor is a bad naming as its job is to deal with an Auth stuff and no more. So I'd suggest renaming it to smth like AuthHttpInterceptor

Reply:
Fixed